src/EditorFeatures/Core/Shared/Utilities/ResettableDelay.cs

```diff
@@ -39,6 +41,16 @@ public ResettableDelay(int delayInMilliseconds, TaskScheduler foregroundTaskSche
             }
         }
 
+        private ResettableDelay()
+        {
+            // create resettableDelay with completed state
+            _delayInMilliseconds = 0;
+            _taskCompletionSource = new TaskCompletionSource<object>();
+            _taskCompletionSource.SetResult(null);
+
+            Reset();
+        }
+
         public Task Task => _taskCompletionSource.Task;
```

Comment:
it seems super weird that we're using this type, which wraps a tcs, instead of just using a tcs directly.

Reply:
sure, you can go and +1 on deleting resetableDelay.